cdap-explore/src/main/java/co/cask/cdap/hive/datasets/DatasetOutputFormat.java

```diff
@@ -67,11 +73,25 @@ public void checkOutputSpecs(FileSystem ignored, JobConf jobConf) throws IOExcep
     private final DatasetAccessor datasetAccessor;
     private final RecordWritable recordWritable;
     private final Type recordType;
+    private Schema recordSchema;
 
     public DatasetRecordWriter(DatasetAccessor datasetAccessor) {
       this.datasetAccessor = datasetAccessor;
       this.recordWritable = datasetAccessor.getDataset();
       this.recordType = recordWritable.getRecordType();
+      if (recordType == StructuredRecord.class) {
+        try {
+          DatasetSpecification datasetSpec = datasetAccessor.getDatasetSpec();
+          String schemaStr = datasetSpec.getProperty(DatasetProperties.SCHEMA);
+          // should never happen, as this should have been checked at table creation
+          if (schemaStr == null) {
+            throw new IllegalStateException("Dataset does not have the schema property.");
+          }
+          recordSchema = Schema.parseJson(schemaStr);
+        } catch (IOException | DatasetManagementException e) {
+          throw new RuntimeException("Unable to look up schema for dataset.", e);
```

Comment:
same with the pattern where we getDatasetAccessor but don't close that on exception..